src/panels/RetinaCapturePanel.ts

```diff
@@ -0,0 +1,169 @@
+            retinaCaptureResult: true,
+        }
+    }
+
+    getInitialState(): InitialState {
+        return {
+            clusterName: this.clusterName,
+            retinaOutput: [this.retinaOutput],
+            allNodes: this.allNodeOutput,
+            selectedNode: "",
+            captureFolderName: this.captureFolderName,
+        };
+    }
+
+    getMessageHandler(): MessageHandler<ToVsCodeMsgDef> {
+        return {
+            runRetinaCapture: (node: string) => this.handleRunRetinaCapture(node),
+            retinaCaptureResult: () => { },
+            getAllNodes: () => { }
+        };
+    }
+
+    private async handleRunRetinaCapture(node: string) {
+        const localCaptureUri = await window.showSaveDialog({
+            defaultUri: Uri.file(this.captureFolderName),
+            saveLabel: "Download",
+            title: "Download Retina File",
+        });
+
+        if (!localCaptureUri) {
+            return;
+        }
+
+        const localCpPath = getLocalKubectlCpPath(localCaptureUri);
+
+        const createPodYaml = `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: node-explorer-${node}
+spec:
+  nodeName: ${node}
+  tolerations:
+    - key: CriticalAddonsOnly
+      operator: Exists
+    - effect: NoExecute
+      operator: Exists
+    - effect: NoSchedule
+      operator: Exists
+  volumes:
+  - name: mnt-captures
+    hostPath:
+      path: /mnt/capture
+  containers:
+  - name: node-explorer
+    image: alpine
+    command: ["sleep", "9999999999"]
+    volumeMounts:
+    - name: mnt-captures
+      mountPath: /mnt/capture
+`  ;
+
+        const applyResult = await longRunning(`Deploying pod to capture ${node} retina data.`, async () => {
+            return await withOptionalTempFile(createPodYaml, "YAML", async (podSpecFile) => {
+                const command = `apply -f ${podSpecFile}`;
+                return await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+            });
+        });
+
+        if (failed(applyResult)) {
+            vscode.window.showErrorMessage(`Failed to apply Pod: ${applyResult.error}`);
+            throw new Error(`Failed to apply Pod: ${applyResult.error}`);
+        }
+        const waitResult = await longRunning(`waiting for pod to get ready node-explorer-${node}.`, async () => {
+            const command = `wait pod -n default --for=condition=ready --timeout=300s node-explorer-${node}`;
+            return await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+        });
+
+        if (failed(waitResult)) {
+            vscode.window.showErrorMessage(`Failed to wait for Pod to be ready: ${waitResult.error}`);
+            return;
+        }
+
+
+        // // kubectl cp 
+        const nodeExplorerResult = await longRunning(`Copy captured data to local host location ${localCpPath}.`, async () => {
+            const cpcommand = `cp node-explorer-${node}:mnt/capture ${localCpPath} --retries 99`;
```

Comment:
I am wondering what is this 99 retries? does it retry 99 times?

Reply:
❤️🙏😊 you will get this answer in the kubectl copy command bug ☕️

https://github.com/kubernetes/kubernetes/issues/60140

Reply:
I think the `--retries` option is only available for `kubectl` versions >= 1.23.0. Are we assuming users will be on that version or later?